Formula/iqtree3.rb

```diff
@@ -0,0 +1,83 @@
+class Iqtree3 < Formula
+  # cite Wong_2025: "https://doi.org/10.32942/X2P62N"
+  desc "Efficient and versatile phylogenomic software by maximum likelihood"
+  homepage "http://www.iqtree.org/"
+  # Use git clone to get submodules
+  url "https://github.com/iqtree/iqtree3.git", branch: "master", shallow: false
+  version "3.0.1"
+  license "GPL-2.0-only"
+  head "https://github.com/iqtree/iqtree3.git", branch: "master", shallow: false
+
+  depends_on "boost" => :build
+  depends_on "cmake" => :build
+  depends_on "eigen" => :build
+  depends_on "googletest" if OS.mac? # For test helpers
+  depends_on "libomp" if OS.mac?
+  depends_on "llvm" if OS.mac?
+  depends_on "zlib"
+  depends_on "lsd2" => :optional
+
+  fails_with gcc: "4"
+  fails_with gcc: "5" do
+    version "5.0"
+    cause "Requires C++17 support"
+  end
+
+  def install
+    # Initialize and update submodules
+    system "git", "submodule", "update", "--init", "--recursive"
+
+    # Set up environment for OpenMP
+    ldflags = "-L#{Formula["libomp"].opt_lib}"
+    cppflags = "-I#{Formula["libomp"].opt_include}"
+
+    # Set environment to prevent FetchContent downloads
+    ENV["FETCHCONTENT_FULLY_DISCONNECTED"] = "ON"
+    ENV["FETCHCONTENT_TRY_FIND_PACKAGE_MODE"] = "ALWAYS"
+    ENV.append "CFLAGS", "-I#{Formula["googletest"].opt_include}/googletest/googletest"
+    args = std_cmake_args + [
+      "-DCMAKE_C_COMPILER=#{ENV.cc}",
+      "-DCMAKE_CXX_COMPILER=#{ENV.cxx}",
+      "-D}CMAKE_CXX_FLAGS=#{ldflags} #{cppflags}",
+      "-DEIGEN3_INCLUDE_DIR=#{Formula["eigen"].opt_include}/eigen3",
+      "-DBUILD_TESTING=OFF",
+      "-DIQTREE_TEST=OFF",
+      "-DFETCHCONTENT_FULLY_DISCONNECTED=ON",
+      "-DZLIB_ROOT=#{Formula["zlib"].opt_prefix}",
+      "-DUSE_CMAPLE=OFF", "-DUSE_LSD2=OFF",
+      "-DFETCHCONTENT_TRY_FIND_PACKAGE_MODE=ALWAYS"
```

Comment:
```suggestion
      "-DUSE_CMAPLE=OFF",
      "-DUSE_LSD2=ON",
      "-DFETCHCONTENT_TRY_FIND_PACKAGE_MODE=ALWAYS",
```

- Turn on `-DUSE_LSD2=ON` to use `brewsci/bio/lsd2`.